content/movie-review-pt1.md

```diff
@@ -0,0 +1,464 @@
+
+## Rust basics
+
+Before we dive into the specifics of a basic Solana program, let's talk about the Rust basics we'll be using throughout this lesson.
+
+### Variables
+
+Variable assignment in Rust happens with the `let` keyword.
+
+```rust
+let age = 33;
+```
+
+Variables in Rust by default are immutable, meaning a variable's value cannot be changed once it has been set. In order to create a variable that we'd like to change at some point in the future, we must make use of the `mut` keyword, which stands for mutable. Defining a variable with this keyword means that the value stored in it can change.
+
+```rust
+// compiler will throw error
+let age = 33;
+age = 34;
+
+// this is allowed
+let mut mutable_age = 33;
+mutable_age = 34;
+```
+The Rust compiler guarantees that when you state a value won’t change, it really won’t change, so you don’t have to keep track of it yourself. This makes your code easier to reason through.
+
+### Structs
+
+A struct, or structure, is a custom data type that lets you package together and name multiple related values that make up a meaningful group. Each piece of data in a struct can be of different types and each has a name associated with it. These pieces of data are called **fields**. They behave similarly to properties in other languages.
+
+```rust
+struct User {
+    active: bool,
+    email: String,
+    age: u64,
+}
+```
+
+To use a struct after we’ve defined it, we create an instance of that struct by specifying concrete values for each of the fields.
+
+```rust
+let mut user1 = User {
+    active: true,
+    email: String::from("test@test.com"),
+    age: 36
+};
+```
+
+To get or set a specific value from a struct, we use dot notation.
+
+```rust
+user1.age = 37;
+```
+
+### Enumerations
+Enumerations (or Enums) are a data struct that allow you to define a type by enumerating its possible variants. An example of an enum may look like:
+
+```rust
+enum LightStatus {
+    On,
+    Off
+}
+```
+
+The `LightStatus` enum has two possible variants in this situation: it's either `On` or `Off`.
+
+You can also embed values into enum variants, similar to adding fields to a struct. 
+
+```rust 
+enum LightStatus {
+    On {
+        color: String
+    },
+    Off
+}
+
+let light_status = LightStatus::On { color: String::from("red") };
+```
+
+In this example, setting a variable to the `On` variant of `LightStatus` requires also setting the value of `color`.
+
+### Match statements
+
+Match statements are a control flow struct that is very similar to a `switch` statement in C/C++. The `match` statement allows you to compare a value against a series of patterns and then execute code based on which pattern matches the value. Patterns can be made of literal values, variable names, wildcards, and more. The match statement must include all possible scenarios, otherwise the code will not compile.
```

Comment:
Think "Match statements are a control flow struct that is very similar to a `switch` statement in C/C++" should be changed to "Match statements are a control flow struct that are very similar to a `switch` statement in C/C++"

Reply:
I think the verb here is referring to the subject "control flow struct" so would be singular, but I could be wrong. Maybe I can reword so it's not even a question